File: app/src/main/java/com/arielfaridja/ezrahi/entities/ReportStatus.java
package com.arielfaridja.ezrahi.entities;

public enum ReportStatus {
    REPORTED(1),
    HANDLED(2);

    private final int value;

    ReportStatus(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static ReportStatus getByValue(int value) {
        switch (value) {
            case 1:
                return REPORTED;
            case 2:
                return HANDLED;
            default:
                return REPORTED;
        }
    }
}

File: app/src/main/java/com/arielfaridja/ezrahi/entities/ReportStatus.kt
package com.arielfaridja.ezrahi.entities

enum class ReportStatus(val value: Int) {
    REPORTED(1),
    HANDLED(2),
    UNKNOWN(-1);

    companion object {
        fun getByValue(value: Int): ReportStatus = when (value) {
            1 -> REPORTED
            2 -> HANDLED
            else -> UNKNOWN
        }
    }
}
